### ISSUE TITLE
Question (and opinion) about object selectors

### ISSUE THREAD

**Issue by umezawatakeshi** [VictoriaMetrics/operator#383]
Why the behavior of [object selectors](https://github.com/VictoriaMetrics/operator/blob/master/docs/quick-start.MD#object-selectors) was changed at [v0.20.0](https://github.com/VictoriaMetrics/operator/releases/tag/v0.20.0) (or https://github.com/VictoriaMetrics/operator/pull/341) ?

--

Suppose that a k8s cluster have multiple tenants which own thier namespaces and the tenants have their own VMAgents, etc. When a new kind of vm-operator-managed objects is added and its corresponding object selector, *all* tenants have to add configuration like

```yaml
  newObjectNamespaceSelector:
    matchLabels:
      tenantName: foo
```

or

```yaml
  newObjectSelector: {}
```

even if the tenants are not interested in those objects in order to accidentaly gather those objects owned by other tenants.

Under the old behavior, only the objects like VMAgents that were interested in other namespaces needed to have configuration like

```yaml
  newObjectNamespaceSelector: {}
```

I think that the new behavior is undesirable in terms of forcing people to write about things they are not interested in.

**Comment by f41gh7** [comment]
Hi, initial intention was to reduce config boilerplate and configuration complexity for users with single vmagent for whole cluster.

But it creates additional problems and now i see, it was wrong decision. 

 So, i want to revert those changes back at 0.21.0 release. 

 And add new spec option - `selectAllScrapes: boolean`, which instructs to threat empty selector as select all. By default, it will be false.
 WDYT?

**Comment by umezawatakeshi** [comment]
> initial intention was to reduce config boilerplate and configuration complexity for users with single vmagent for whole cluster.

I see... Certainly, I agree that the new behavior is preferable for a simple single-owner cluster.

> And add new spec option - selectAllScrapes: boolean, which instructs to threat empty selector as select all. By default, it will be false.

I think it's not a bad idea :+1:. Will other kind of specs (VMAlerts, VMAlertmanagers, VMAuths) have similar option?

Just to confirm, the option specifies the bahavior only in case both <code><i>something</i>NamespaceSelector</code> and <code><i>something</i>Selector</code> are nil and the behavior in other cases will be same as the current behavior, right?

**Comment by f41gh7** [comment]
> 
> I think it's not a bad idea +1. Will other kind of specs (VMAlerts, VMAlertmanagers, VMAuths) have similar option?
> 

 Yes, all objects will have such option - `selectAllByDefault` with default value of false.

> Just to confirm, the option specifies the bahavior only in case both `somethingNamespaceSelector` and `somethingSelector` are nil and the behavior in other cases will be same as the current behavior, right?
  
 Yep, its correct.

**Comment by f41gh7** [comment]
Changes was included to the 0.21.0 release.
